### PR TITLE
add some tests for eth-backing

### PR DESCRIPTION
**Pull request by readlnh**

### CODE REVIEW

File: srml/eth-backing/src/tests.rs [hunk 1]
```diff
@@ -55,3 +57,113 @@ fn verify_parse_token_redeem_proof() {
...
+				gas_used: 0x78bd9a.into(),
+				gas_limit: 0x79d4fe.into(),
+				difficulty: 0x75e5a3ef_u64.into(),
+				seal: vec![rlp::encode(&mixh), rlp::encode(&nonce)],
+				hash: Some(H256::from(hex!("f3cc3fab1b6cae48660a36839630c350bace54156d57ee3c62c6113d4b7d82b1"))),
+			};
+
+			assert_ok!(EthRelay::init_genesis_header(&header, 0x68de130d2c02a8_u64));
+
+			let expect_account_id = <Test as Trait>::DetermineAccountId::account_id_for(&hex!("2a92ae5b41feba5ee68a61449c557efa9e3b894a6461c058ec2de45429adb44546")).ok().unwrap();
```
Comment (AurevoirXavier) [inline srml/eth-backing/src/tests.rs]: `ok()` does nothing here.

File: srml/eth-backing/src/tests.rs [hunk 2]
```diff
@@ -55,3 +61,191 @@ fn verify_parse_token_redeem_proof() {
...
+			assert_ok!(
+				staking::Call::<Test>::bond(
+					AccountId32::from([1; 32]),
+					StakingBalances::Ring(1),
+					RewardDestination::Controller,
+					0).dispatch(Origin::signed(expect_account_id.clone()))
+			); 	
+		 	assert_ok!(EthBacking::redeem_deposit(Origin::signed(id1.clone()), proof_record.clone()));
+
+		 	assert_eq!(Ring::free_balance(&expect_account_id), 1234000000000 + 1);
```
Comment (HackFisher) [inline srml/eth-backing/src/tests.rs]: TODO: Assert deposit Item and active ring

File: srml/eth-backing/src/tests.rs (hunk 1 above)
Comment (HackFisher) [reply srml/eth-backing/src/tests.rs]: Fixed in https://github.com/darwinia-network/darwinia/pull/172/commits/07694a353093cad2723c3f08001d702c4fe8664b

File: srml/eth-backing/src/tests.rs (hunk 2 above)
Comment (HackFisher) [reply srml/eth-backing/src/tests.rs]: Fixed in https://github.com/darwinia-network/darwinia/pull/172/commits/07694a353093cad2723c3f08001d702c4fe8664b

**Pull request merged**